### ISSUE TITLE
[FEA] `PredictForest` to accept a `Merlin Model`

### ISSUE THREAD

**Issue by radekosmulski** [NVIDIA-Merlin/systems#144]
# 🚀 Feature request

Currently, when passed a `merlin.models.xgb.XGBoost`, `PredictForest` fails with the following error:
```
ValueError: Model type not supported. <class 'merlin.models.xgb.XGBoost'> Must be one of: {'lightgbm.Booster', 'xgboost.XGBModel', 'cuml.ensemble.RandomForestRegressor', 'cuml.ensemble.RandomForestClassifier', 'lightgbm.LGBMModel', 'sklearn.ensemble.RandomForestClassifier', 'sklearn.ensemble.RandomForestRegressor', 'xgboost.Booster'}
```

It would be great if a user would be able to pass a `merlin.models.xgb.XGBoost` instance directly to `PredictForest`

## Motivation

Ideally, most users using our high-level API will not even realize that `model.booster` exists, so when confronted with this error they might be quite confused.

Also, it is great that we support serving of all these other models, but probably for consistency's sake, and for ease of use, we should aim for the `Merlin Models` API to accept our models wherever feasible (while being able to serve all these other models as well).

**Comment by viswa-nvidia** [comment]
@EvenOldridge , do you think this should get added to the scope for 22.08 ?

**Comment by radekosmulski** [comment]
@benfred implemented this in #146
